stan/math/prim/fun/cholesky_corr_free.hpp

```diff
@@ -33,6 +33,22 @@ auto cholesky_corr_free(const T& x) {
   return z;
 }
 
+/**
+ * Overload of `cholesky_corr_free()` to untransform each matrix
+ * in a standard vector.
+ * @tparam T A standard vector with with a `value_type` which inherits from
+ *  `Eigen::MatrixBase`.
+ * @param x The standard vector to untransform.
+ */
+template <typename T, require_std_vector_t<T>* = nullptr>
+auto cholesky_corr_free(const T& x) {
```

Comment:
Will these fail on an empty standard vectors because `x[0]` is not defined?  We can declare as follows legally and people do this to swap in and out variables (usually with the size being some data object).

```
array[0] simplex[5] a;
```

Reply:
This will pass even if `x` is never filled. With your idea below we don't need this directly anymore, but keep in mind the `x[0]` here is done at compile time and only asks for the return type of `x[0]`, which the std::vector<T> knows is `T`.

Reply:
Nice.  That's what I was asking.